src/main/java/app/model/Map.java

```diff
@@ -257,4 +257,27 @@ private boolean clearSpot(Vector v)
         }
         return true;
     }
+
+
+    public void capture(Agent currentAgent)
+    {
+        if(currentAgent.getTeam() == Team.GUARD)
```

Comment:
why not reverse this to make a clean exit?

```java
if(currentAgent.getTeam() != Team.GUARD)
    return;
```

This removes one level of nesting in our function